### DISCUSSION TITLE
是否有办法在Umi中让IntlProvider自动包裹每个createRoot创建的DOM？

### DISCUSSION THREAD

**Discussion by leslie555** [umijs/umi#12529] [Q&A]
正常来说国际化的Provider会包裹root下的组件， 但是我在项目中使用了logicflow来画图，logicflow里面的node使用了react-dom的createRoot来创建元素，root.render渲染的组件无法消费`@umijs/max`下提供的方法. 我尝试了手动给render的组件包裹了IntlProvider，然后render的组件就可以使用国际化相关的钩子。 但是手写一个个的Provider太复杂。 请教有没有办法让render的组件和入口的Provider保持一致。

![image](https://github.com/umijs/umi/assets/40735433/7b2d3eea-a8e6-4915-af99-a3aa61af5bb1)

**Comment by fz6m** [comment]
我理解这个问题属于 logicflow 在 `createRoot` 时需要加一个包裹 context 的问题，要从 logicflow 上解决，比如看看这个库能不能配置，或者 Patch ，PR 代码等方式来解决。

**Reply by leslie555** [reply]
这个root.render是我自己编写的代码， 我需要在logicFlow的画布上面添加点击事件，而这里是点击之后出现drawer, 但是这个drawer的children是项目中的公用组件， 这个公用的组件使用了useModel这个hook， 要使用useModel需要添加什么Provider啦？

**Reply by fz6m** [reply]
umi plugins 提供的功能（如 model / dva 等），不可以在第三方包和上下文内使用，你这里创建了一个新的 root 挂载，相当于一个独立隔离的环境，是用不了 umi 的上下文和插件的。

尝试把节点 portal 传送过去吧，从而共享上下文，以下库供参考：

 - https://github.com/tajo/react-portal

 - https://github.com/alex-cory/react-useportal

 - https://github.com/httptoolkit/react-reverse-portal

 - https://github.com/pmndrs/tunnel-rat
